control-plane/src/internal_dns.rs

```diff
@@ -28,24 +26,26 @@ pub async fn get_ip_for_host_with_dns_resolver(
     dns_resolver: &AsyncDnsResolver,
     host: &str,
     port: u16,
-) -> error::Result<Option<SocketAddr>> {
-    #[cfg(feature = "enclave")]
+) -> error::Result<SocketAddr> {
+    use crate::error::ServerError;
+    use rand::prelude::IteratorRandom;
+
     let addr = dns_resolver
         .lookup_ip(host)
         .await?
         .iter()
         .choose(&mut rand::thread_rng())
         .map(|ip| SocketAddr::new(ip, port));
 
-    Ok(addr)
+    match addr {
+        Some(addr) => Ok(addr),
+        None => Err(ServerError::DNSNotFound),
+    }
```

Comment:
could use ok_or here
```suggestion
    addr.ok_or(ServerError::DNSNotFound)
```